### PR TITLE
Registry features added

### PR DESCRIPTION
**Pull request by nupurgunwant**
This pull request completes the following functions:

- `getInfo("mine")` to find information about a mine

- `getData("mine")` to find names of all data sets corresponding to a mine

- `getMines("oragnism")` to extract the the mines to which the organism belongs

Also, to make this possible `requests` and `yaml` libraries of python were used which are included in `requirements.txt`. Thus, an additional step to be done by the user now is run `pip install -r requirements.txt` after running `python setup.py install`. This can be updated in the READme file if this PR looks fine.
Also, I haven't made changes in the travis files for including the above libraries(requests and yaml), will do so after I write the tests.

PS: the corresponding tutorial is the latest PR in [docs](https://github.com/intermine/intermine-ws-python-docs)

### CODE REVIEW

**Comment by julie-sullivan** [comment]
A couple questions:

1. You told me about checkstyle for Python a few weeks ago. Was that something we were interested in?

2. Why is registry at the top level and not in the `intermine` directory? Is that because it's sort of "special"?

**Comment by nupurgunwant** [comment]
Hi @julie-sullivan and  @rachellyne , 
1. Yes, I suggested using pep8speaks for Python. I haven't used it uptill now but I will surely check on it
2. You are right, it is in fact similar to all other files and should be inside the intermine folder. I have made the change in its location. 

Looking forward to your review.

**Comment by julie-sullivan** [comment]
Steps to test:

```

  * git clone https://github.com/nupurgunwant/intermine-ws-python.git
  * cd intermine-ws-python
  * python setup.py install
  * pip install -r requirements.txt
  * python(this will open the interactive mode of python, make sure the
    version is >=3)
  *  >>> from intermine import registry
  *  >>>registry.getInfo("flymine")
  *  >>>registry.getMines("D. melanogaster")
  *  >>>registry.getData("flymine")
```

File: intermine/registry.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    print("InterMine Version: " + dict["instance"]["intermine_version"])
+    print("Organisms: "),
+    for organism in dict["instance"]["organisms"]:
+        print(organism),
+    print("Neighbours: "),
+    for neighbour in dict["instance"]["neighbours"]:
+        print(neighbour),
+
+def getData(x):
+    link = "http://www." + x + ".org/" + x + "/service"
```
Comment (yochannah) [inline intermine/registry.py]: This works great for some intermines, but not all InterMines follow this url schema - e.g. not http, running on subdomains (like http://maizemine.rnet.missouri.edu:8080/maizemine ), or even something like bovinemine, on http://bovinegenome.org/bovinemine

**Comment by nupurgunwant** [comment]
Hi @yochannah , Thanks for the review.

1. You are absolutely right about the change required in the URL. To solve it, I used the URL from `getInfo(mine)` function instead(like you suggested). I guess its working fine now, but you do let me know if it can be improved.

2. I have also added error messages ie `getData('kittenmine')` now returns **No such mine available**, similarly for the other two functions.

3. About the _suggesting feature_ ie suggesting use of D. melanogaster instead of Drosophila melanogaster, I am sure this is a good thing that can be added. I will surely work on it in the coming time.

**Comment by yochannah** [comment]
wow, that was fast! re-tried the tricky bits and they're all good now. @julie-sullivan happy to merge this if you are

**Pull request merged**